consensus/src/transaction_shuffler/fairness/conflict_key/entry_fun.rs

```diff
@@ -14,21 +16,29 @@ pub enum EntryFunKey {
     Exempt,
 }
 
+impl From<(&ModuleId, &IdentStr)> for EntryFunKey {
+    fn from((module_id, function): (&ModuleId, &IdentStr)) -> Self {
+        if module_id.address().is_special() {
+            // Exempt framework modules
+            Self::Exempt
+        } else {
+            // n.b. Generics ignored.
+            Self::EntryFun {
+                module: module_id.clone(),
+                function: function.to_owned(),
+            }
+        }
+    }
+}
+
 impl ConflictKey<SignedTransaction> for EntryFunKey {
```

Comment:
I am unsure of the role of `ConflictKey` here for consensus. We can discuss.

Reply:
my understanding it that it is used by aptos-consensus to order/select the transactions while constructing a block to avoid conflicts as much as possible.
Here is the comment for it in code-base
```
/// `ConflictKey::extract_from(txn)` extracts a key from a transaction. For example,
/// `TxnSenderKey::extract_from(txn)` returns the transaction sender's address. The key is used by
/// the shuffler to determine whether two close by transactions conflict with each other.
///
/// `ConflictKey::conflict_exempt(&key)` returns if this specific key is exempt from conflict.
/// For example, we can exempt transaction sender 0x1, so that consecutive transactions sent by
/// 0x1 are not seen as a conflict by the shuffler.
```
This feature is used to reorder the existing transactions avoiding conflicts as much as possible and then build the block from the ordered set.
This is aptos internal logic and in smr-moonshot we are not using it.